examples/example.c

```diff
@@ -4,18 +4,36 @@
  */
 
 #include "example_host.h"
+#ifdef STATICALLY_LINKED_EXAMPLE
 #include "example_vm/example_vm.h"
+#endif
 
 #include <evmc/helpers.h>
+#include <evmc/loader.h>
 
 #include <inttypes.h>
 #include <stdio.h>
 
-int main()
+int main(int argc, char* argv[])
 {
+#ifdef STATICALLY_LINKED_EXAMPLE
+    (void)argc;
+    (void)argv;
     struct evmc_instance* vm = evmc_create_example_vm();
+    if (!vm)
+        return -1;
     if (!evmc_is_abi_compatible(vm))
         return -1;
+#else
+    char* config_string = (argc > 1) ? argv[1] : "example-vm.so";
+    enum evmc_loader_error_code error_code;
+    struct evmc_instance* vm = evmc_load_and_configure(config_string, &error_code);
+    if (!vm)
+    {
+        printf("Loading error: %d\n", error_code);
+        return -1;
```

Comment:
```suggestion
        return error_code;
```

Reply:
`error_code` are positive, ok with that? (should be fine as it is non-zero).

Reply:
Success is 0. The rest is positive. I think this is ok from OS point of view. Even better than negative I believe, but not expert here.